src/Common/Scheduler/Workload/WorkloadEntityConfigStorage.cpp

```diff
@@ -0,0 +1,187 @@
+{
+    refreshEntities(config);
+}
+
+WorkloadEntityConfigStorage::OperationResult WorkloadEntityConfigStorage::storeEntityImpl(
+    const ContextPtr &,
+    WorkloadEntityType,
+    const String &,
+    ASTPtr,
+    bool,
+    bool,
+    const Settings &)
+{
+    // Config storage is read-only - entities come from config, not SQL. This function should not be called
+    return OperationResult::Failed;
+}
+
+WorkloadEntityConfigStorage::OperationResult WorkloadEntityConfigStorage::removeEntityImpl(
+    const ContextPtr &,
+    WorkloadEntityType,
+    const String &,
+    bool)
+{
+    // Config storage is read-only - entities come from config, not SQL. This function should not be called
+    return OperationResult::Failed;
+}
+
+void WorkloadEntityConfigStorage::refreshEntities(const Poco::Util::AbstractConfiguration & config)
+{
+    LOG_DEBUG(log, "Refreshing workload entities from configuration");
+
+    std::vector<std::pair<String, ASTPtr>> new_entities;
+
+    // Load resources first (workloads may reference them)
+    if (config.has("predefined_resources"))
+    {
+        Poco::Util::AbstractConfiguration::Keys resource_names;
+        config.keys("predefined_resources", resource_names);
+
+        for (const auto & resource_name : resource_names)
+        {
+            String config_key = "predefined_resources." + resource_name;
+            String sql = config.getString(config_key, "");
+            
+            if (sql.empty())
+            {
+                LOG_WARNING(log, "Resource '{}' has empty SQL configuration", resource_name);
+                continue;
+            }
+
+            if (auto ast = parseEntityFromConfig(WorkloadEntityType::Resource, resource_name, sql))
+                new_entities.emplace_back(resource_name, ast);
+        }
+    }
+
+    // Load workloads after resources
+    if (config.has("predefined_workloads"))
+    {
+        Poco::Util::AbstractConfiguration::Keys workload_names;
+        config.keys("predefined_workloads", workload_names);
+
+        for (const auto & workload_name : workload_names)
+        {
+            String config_key = "predefined_workloads." + workload_name;
+            String sql = config.getString(config_key, "");
+            
+            if (sql.empty())
+            {
+                LOG_WARNING(log, "Workload '{}' has empty SQL configuration", workload_name);
+                continue;
+            }
+
+            if (auto ast = parseEntityFromConfig(WorkloadEntityType::Workload, workload_name, sql))
+                new_entities.emplace_back(workload_name, ast);
+        }
+    }
+
+    // Update entities in memory and notify subscribers
+    setAllEntities(new_entities);
+
+    LOG_DEBUG(log, "Loaded {} workload entities from configuration", new_entities.size());
+}
+
+ASTPtr WorkloadEntityConfigStorage::parseEntityFromConfig(WorkloadEntityType entity_type, const String & entity_name, const String & sql)
```

Comment:
This function is not needed, reuse code from workload entity keeper storage class. Move the necessary code into common base class to reuse it (i.e. add static method for parsing semicolon separated list of SQL statements from string)